CI runs out of space while building servo

The merge queue failed in https://github.com/servo/servo/actions/runs/20469836410/ because the linker ran out of space while building servo.

```
  = note: /usr/bin/ld: final link failed: No space left on device
          collect2: error: ld returned 1 exit status
```

Unlike https://github.com/servo/servo/issues/35179 this seems to affect self-hosted runners. At least I think so, https://github.com/servo/servo/actions/runs/20469836410/job/58822351380 says ` is_self_hosted=true`.

--- a/.github/workflows/linux.yml
+++ b/.github/workflows/linux.yml
@@ -272,6 +272,13 @@ jobs:
         with:
           ref: ${{ github.event.pull_request.head.sha }}
           fetch-depth: 1
+      - name: Free Disk Space (Ubuntu)
+        uses: jlumbroso/free-disk-space@main
+        if: ${{ runner.environment != 'self-hosted' }}
+        with:
+          tool-cache: false
+          large-packages: false
+          swap-storage: false
       - name: Set LIBCLANG_PATH env # needed for bindgen in mozangle
         run: echo "LIBCLANG_PATH=/usr/lib/llvm-14/lib" >> $GITHUB_ENV
       - name: Setup Python

